holidays/utils.py

```diff
@@ -13,114 +13,370 @@
+    "MK": "north_macedonia",
+    "MP": "northern_mariana_islands",
+    "NO": "norway",
+    "PK": "pakistan",
+    "PA": "panama",
+    "PY": "paraguay",
+    "PE": "peru",
+    "PH": "philippines",
+    "PL": "poland",
+    "PT": "portugal",
+    "PR": "puerto_rico",
+    "RO": "romania",
+    "RU": "russia",
+    "SM": "san_marino",
+    "SA": "saudi_arabia",
+    "RS": "serbia",
+    "SG": "singapore",
+    "SK": "slovakia",
+    "SI": "slovenia",
+    "ZA": "south_africa",
+    "ES": "spain",
+    "SE": "sweden",
+    "CH": "switzerland",
+    "TW": "taiwan",
+    "TH": "thailand",
+    "TN": "tunisia",
+    "TR": "turkey",
+    "UA": "ukraine",
+    "AE": "united_arab_emirates",
+    "GB": "united_kingdom",
+    "US": "united_states",
+    "UM": "united_states_minor_outlying_islands",
+    "UY": "uruguay",
+    "UZ": "uzbekistan",
+    "VE": "venezuela",
+    "VN": "vietnam",
+    "VI": "united_states_virgin_islands",
+    "ZM": "zambia",
+    "ZW": "zimbabwe",
+}
+market_to_module = {
+    # List of all financial market codes we support and the submodule name
+    # The code itself must be the name of the class in that submodule
+    # ISO 10383 market identifier codes (MIC):
+    "XNYS": "ny_stock_exchange",
+    # Pseudocodes (no ISO 10383 exists):
+    "ECB": "european_central_bank",
+    # Aliases:
+    "NYSE": "ny_stock_exchange",
+    "TAR": "european_central_bank",
+}
+
+
+class Holidays(HolidayBase):
+    """
+    This class instantiates public holidays for a single country or financial
+    market. It maintains backwards compatibility with the original
+    load-all-objects architecture while dramatically improving loading times by
+    implementing lazy loading of the country or market, which is performed by
+    the function :py:func:`country_holidays` or :py:func:`financial_holidays`.
+
+    It does this by "inheriting" (i.e. adding) all the attributes and methods
+    of the country- or market-specific subclass found in its file.
+
+    TODO: Once the original load-all-objects architecture is fully deprecated,
+    this class can be simplified/rearchitected.  For example, we can get rid of
+    the bloat in submodules and have a single class per country/market.
+
+    As a side benefit, all public holiday objects loaded correctly are of this
+    :py:class:`Holidays` class (as opposed to a country- or market-specific
+    subclass), and therefore easy to spot in development and fully documented.
+
+    This class should not be instantiated directly (or lazy-loading won't take
+    place), rather the public holiday :py:class:`Holidays` object should be
+    instantiated by calling one of the two specialized functions, who also
+    perform argument validation and provide documentation, as follows:
+
+    >>> from holidays import country_holidays
+    >>> us_holidays = country_holidays('US')
+
+    >>> from holidays import financial_holidays
+    >>> nyse_holidays = financial_holidays('XNYS')
+    """
+
+    def __init__(
```

Comment:
`expand` is supported in the public API to maintain 100% backwards compatibility. It's streamlined away internally since it's implemented in exactly zero lines of code for zero countries or financial markets.